diff --git a/config.def.h b/config.def.h
index 9efa774..41cd985 100644
--- a/config.def.h
+++ b/config.def.h
@@ -77,6 +77,7 @@ static const Key keys[] = {
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
+	{ MODKEY, 	                XK_f,      fullscreen,     {0} },
 	{ MODKEY,                       XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
diff --git a/dwm.c b/dwm.c
index 1443802..418711f 100644
--- a/dwm.c
+++ b/dwm.c
@@ -199,6 +199,7 @@ static void sendmon(Client *c, Monitor *m);
 static void setclientstate(Client *c, long state);
 static void setfocus(Client *c);
 static void setfullscreen(Client *c, int fullscreen);
+static void fullscreen(const Arg *arg);
 static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
 static void setup(void);
@@ -1506,6 +1507,22 @@ setfullscreen(Client *c, int fullscreen)
 	}
 }
 
+void
+fullscreen(const Arg *arg)
+{
+	if (selmon->sel && selmon->showbar) {
+		if (selmon->sel->isfloating) {
+			selmon->sel->isfloating = !selmon->sel->isfloating;
+			resize(selmon->sel, selmon->sel->x, selmon->sel->y,
+				       selmon->sel->w, selmon->sel->h, 0);
+		}
+		setlayout(&((Arg) { .v = &layouts[2] }));
+	} else {
+		setlayout(&((Arg) { .v = &layouts[0] }));
+	}
+	togglebar(arg);
+}
+
 void
 setlayout(const Arg *arg)
 {
